services/placement/algo/sharded_helper.go

```diff
@@ -36,7 +36,7 @@ type PlacementHelper interface {
 	PlaceShards(shards []shard.Shard, from services.PlacementInstance) error
 
 	// GeneratePlacement generates a placement
-	GeneratePlacement() services.ServicePlacement
+	GeneratePlacement(includeEmptyInstance bool) services.ServicePlacement
```

Comment:
Perhaps it's better to make it an enum so it's easier to understand at callsite:
```
type includeInstanceType int

const (
  includeEmpty includeInstanceType = iota
  nonEmptyOnly
)

GeneratePlacement(instanceType includeInstanceType) service.ServicePlacement
```
Then it's easier to understand what it means at callsite
```
GeneratePlacement(includeEmpty)
```